### PR TITLE
Cleanup the code

### PR DESCRIPTION
**Pull request by Zezombye**
Added a global "switching hero" rule that handles switching and resets the player properties (abilities, health, etc).

This rule also solves the echo problem with a player "hero" variable, and I've merged duplicate rules when possible (when there was no wait). The remaining duplicate rules have a wait and therefore likely depend on the event hero behavior, so they will need to be converted more carefully.

Cleaned up various things, removed some gotos, merged strings, and fixed some minor bugs (eg hanzo doing 3x damage instead of 2x).

### CODE REVIEW

File: src/general/global.opy [hunk 1]
```diff
@@ -4,6 +4,7 @@
 subroutine CheckStatus
 
 
+#this is useless right?
```
Comment (Mitcheljager) [inline src/general/global.opy:7]: Why? It's used in several places. I think for Winston and for Mei only at the moment

File: src/heroes/dva/baby.opy [hunk 2]
```diff
@@ -3,16 +3,8 @@
...
-
-
-rule "[Echo] Set baby D.va Ult charge to 0":
-    @Event eachPlayer
-    @Hero echo
-    @Condition eventPlayer.getHeroOfDuplication() == Hero.DVA
+    @Condition eventPlayer.hero == Hero.DVA
     @Condition eventPlayer.isInAlternateForm()
 
+    #don't you already start with 0 ult charge as baby?
```
Comment (Mitcheljager) [inline src/heroes/dva/baby.opy:9]: There was a problem with one of the subroutines that checks what your previous Ult charge was for the Tank reduced Ult charge passive. Baby d.va would have the ult charge she had as big d.va

File: src/main.opy [hunk 3]
```diff
@@ -2,6 +2,61 @@
 
 #!include "settings.opy"
 
+
+playervar hero
```
Comment (Mitcheljager) [inline src/main.opy:6]: I believe there already is a variable that also sets the current hero `HeroActive`

File: src/general/global.opy (hunk 1 above)
Comment (Zezombye) [reply src/general/global.opy:7]: It is a noop, it just does a bunch of ifs then "return".

File: src/main.opy (hunk 3 above)
Comment (Zezombye) [reply src/main.opy:6]: Probably ye, I saw something about this, however I saw that HeroActive can be set to "null".

File: src/general/global.opy (hunk 1 above)
Comment (Mitcheljager) [reply src/general/global.opy:7]: I thought the same, but didn't make it so I assumed it worked, but I will do some testing to see if really does work.

Comment (Mitcheljager) [reply src/general/global.opy:7]: Update; it doesn't work

**Pull request merged**